support/ci/shared.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+set -eou pipefail
+
+#
+# Install core rust toolchain
+# defaults to "stable"
+#
+install_rust_toolchain() {
+  toolchain="${1:-stable}"
+
+  echo "--- :rust: Installing rustup"
+  curl https://sh.rustup.rs -sSf | sh -s -- --no-modify-path -y
```

Comment:
It'd be nice to avoid the network call if we already have `rustup` and `cargo`:
```bash
  if command -v rustup && command -v cargo; then
    echo "--- :rust: $(rustup --version) already installed"
	else
    echo "--- :rust: Installing rustup"
    curl https://sh.rustup.rs -sSf | sh -s -- --no-modify-path -y
    source "$HOME"/.cargo/env
  fi
```